org.sailfishos.usermanager.xml

```diff
@@ -43,6 +43,19 @@
     <method name="currentUser">
         <arg direction="out" type="u" name="uid"/>
     </method>
+    <method name="usersGroups">
+        <arg direction="in" type="u" name="uid"/>
+        <arg direction="out" type="a(s)" name="groups"/>
+        <annotation name="org.qtproject.QtDBus.QtTypeName.Out0" value="QStringList"/>
+    </method>
+    <method name="addToGroup">
+        <arg direction="in" type="u" name="uid"/>
+        <arg direction="in" type="s" name="group"/>
```

Comment:
Why group name and not gid? For consistency with users, if nothing else. Besides, renaming of groups is easier to imagine than changing gids, which makes gid a better candidate for a group identifier. Slightly more efficient, too.

Reply:
I'm thinking UI would like to use strings here to enable/disable services for users, it does not know the gids but the names can be specified, some are already in use. Opinions @Tomin1 ?

Reply:
I wouldn't be thinking too much about UI at this point. UI can (and likely will) change in the future, D-Bus API is going to stay, and the cost of making a mistake is higher at lower level.

Reply:
Well, this is actually quite interesting question. We have specified group names (see sailfish-setup) that we use but gids don't have any meaning really so from that point of view it's always an extra lookup to know how they match. Uids are much more meaningful IMO. Also we are not planning to allow creation or removal of groups (other than user's own group while creating or removing user) through this service and personally I don't think that's going to change in future either but who knows of course. I would say that for groups the names are fine but I'll leave that to you to decide. Settings plugin can adjust to either model as needed.

Reply:
I'd say: user needs to be identified in a manner that is always resolvable in deterministic manner - and we are explicitly planning for cases like renaming user, or?.  The group bit here is just adjusting attributes of user which needs to hold water during the time of the change (and renaming group names without changing gid - what would be the use for that?). If the typical use case is adding groups by name, using gid would mean N x lookup code at client side vs. one place at service. I would prefer having potential hazard / complexity in one place.

Reply:
Yeah, cleanly renaming groups in our setup is just as impossible as changing numeric group ids, so I guess it doesn't really matter. When I have a choice, I usually prefer to use numbers rather than strings (because CPUs are much better suited for working with numbers, numbers have no issues with leading/trailing blank characters, upper/lower case and so on) but I guess it's up to whoever is going to maintain this code. Which is not me :)

Reply:
Using gids would be more robust but in this case the extra steps/work needed  from UI to use them is not worth it, keeping this as is.